File: Colas/Cola de prioridad descendente/Cola_Descendente.cpp
#include <iostream>
using namespace std;

#define MAXSIZE 5
int queue[MAXSIZE];
int count = 0;

void insertar() {
    if (count == MAXSIZE) {
        cout << "\nDESBORDAMIENTO (OVERFLOW)\n";
        return;
    }

    int elemento;
    cout << "\nIngrese el elemento: ";
    cin >> elemento;

    int i = count - 1;
    while (i >= 0 && queue[i] < elemento) {
        queue[i + 1] = queue[i];
        i--;
    }
    queue[i + 1] = elemento;
    count++;

    cout << "\nElemento insertado correctamente.\n";
}

void eliminar() {
    if (count == 0) {
        cout << "\nSUBDESBORDAMIENTO (UNDERFLOW)\n";
        return;
    }

    cout << "\nElemento eliminado (mayor prioridad): " << queue[0] << "\n";
    for (int i = 1; i < count; i++) {
        queue[i - 1] = queue[i];
    }
    count--;
}

void mostrar() {
    if (count == 0) {
        cout << "\nLa cola está vacía.\n";
        return;
    }

    cout << "\nElementos (orden descendente):\n";
    for (int i = 0; i < count; i++)
        cout << queue[i] << " ";
    cout << endl;
}

int main() {
    int opcion;
    do {
        cout << "\n******** COLA DE PRIORIDAD DESCENDENTE ********\n";
        cout << "1. Insertar\n2. Eliminar\n3. Mostrar\n4. Salir\n";
        cout << "Ingrese su opción: ";
        cin >> opcion;

        switch (opcion) {
            case 1: insertar(); break;
            case 2: eliminar(); break;
            case 3: mostrar(); break;
            case 4: cout << "\nSaliendo...\n"; break;
            default: cout << "\nOpción inválida.\n";
        }
    } while (opcion != 4);
    return 0;
}

File: Colas/Cola de prioridad descendente/Cola_Descendente.py
MAXSIZE = 5
queue = []

def insertar():
    if len(queue) == MAXSIZE:
        print("OVERFLOW")
        return
    e = int(input("Ingrese el elemento: "))
    queue.append(e)
    queue.sort(reverse=True)
    print("Elemento insertado.")

def eliminar():
    if not queue:
        print("UNDERFLOW")
        return
    print("Elemento eliminado:", queue.pop(0))

def mostrar():
    if not queue:
        print("Vacía")
    else:
        print("Cola descendente:", queue)

while True:
    print("\n1.Insertar 2.Eliminar 3.Mostrar 4.Salir")
    op = input("Opción: ")
    if op == "1": insertar()
    elif op == "2": eliminar()
    elif op == "3": mostrar()
    elif op == "4": break